src/duqtools/schema/_dimensions.py

```diff
@@ -132,8 +132,13 @@ def npfunc(self,
             npfunc = getattr(np, self.operator)
 
         # copyto is different, and does not like scalars
-        if self.operator == 'copyto' and not isinstance(data, np.ndarray):
-            return data
+        #if self.operator == 'copyto' and not isinstance(data, np.ndarray):
+        #    return data
+        if self.operator == 'copyto':
+            if isinstance(value, type(data)):
+                return value
+            elif not isinstance(data, np.ndarray):
+                return data
```

Comment:
I think this makes it a bit clearer to me:

```suggestion
        if self.operator == 'copyto' and not isinstance(data, np.ndarray):
            if isinstance(value, type(data)):
                return value
            elif else:
                return data
```